docs/en/modules/qrcodegen.md

```diff
@@ -0,0 +1,70 @@
+# QR Code Generator Module
+| Since  | Origin / Contributor  | Maintainer  | Source  |
+| :----- | :-------------------- | :---------- | :------ |
+| 2018-10-20 | https://github.com/nayuki/QR-Code-generator.git integrated by [Tom Sutcliffe](https://github.com/tomsci) | [Tom Sutcliffe](https://github.com/tomsci) | [qrcodegen.c](../../../components/modules/qrcodegen.c)|
+
+This module wraps the [QR Code](https://en.wikipedia.org/wiki/QR_code) Generator API written in C by https://github.com/nayuki/QR-Code-generator.git for producing a QR Code.
+
+## qrcodegen.encodeText()
+Generates a QR Code from a text string. In the most optimistic case, a QR Code at version 40 with low ECC can hold any UTF-8 string up to 2953 bytes, or any alphanumeric string up to 4296 characters, or any digit string up to 7089 characters.
+
+#### Syntax
+`qrcodegen.encodeText(text, minVer, maxVer, ecl, mask, boostEcl)`
```

Comment:
The high degree of freedom with optional positional parameters would call for a rather complex formal syntax of the API. E.g.:
```
`qrcodegen.encodeText(text[, [minVer][, [maxVer][, [ecl][, [mask][, [boostEcl]]]]]])`
```
How about an optional table?

Reply:
I think it'd probably be `qrcodegen.encodeText(text[, minVer[, maxVer[, ecl[, mask[, boostEcl]]]]])` wouldn't it? But yes, it's a bit clunky to write. I was on the fence as to whether to make an options table but you end up having to write a bit more boilerplate without the `luaL_opt*` fns to help so I didn't bother. Maybe we should add some more helper fns for things like options tables to some shared file? Otherwise I'll be duplicating the `check_optint/check_optbool` code from `http.c` in order to do thorough argument validation.

Reply:
:smile: I understood the code the way that `qrcodegen.encodeText("lorem ipsum", , 4)` is accepted.

> Maybe we should add some more helper fns for things like options tables to some shared file?

Yes, agree to that. I too felt to reinvent the wheel more than once in the past. Each time with a different look of course. Though I can't really propose a good solution atm. Maybe just a simple `components/modules/common.c`?